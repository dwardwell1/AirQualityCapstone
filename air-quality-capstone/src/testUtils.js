import React from 'react';
import UserContext from './auth/UserContext';

const demoUser = {
	username: 'testuser',
	email: 'test@test.net',
	default_locale: '95926',
	alerts: 5
};

const UserProvider = ({ children, currentUser = demoUser }) => (
	<UserContext.Provider value={{ currentUser }}>{children}</UserContext.Provider>
);

export { UserProvider };
