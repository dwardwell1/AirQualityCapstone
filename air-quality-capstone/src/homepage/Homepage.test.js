import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router';
import Home from './Homepage';
import { UserProvider } from '../testUtils';

it('matches snapshot', function() {
	const { asFragment } = render(
		<MemoryRouter>
			<UserProvider>
				<Home />
			</UserProvider>
		</MemoryRouter>
	);
	expect(asFragment()).toMatchSnapshot();
});

it('matches snapshot when logged out', function() {
	const { asFragment } = render(
		<MemoryRouter>
			<UserProvider currentUser={null}>
				<Home />
			</UserProvider>
		</MemoryRouter>
	);
	expect(asFragment()).toMatchSnapshot();
});

test('Shows login option if no user', () => {
	const { getByText } = render(
		<MemoryRouter>
			<UserProvider currentUser={null}>
				<Home />
			</UserProvider>
		</MemoryRouter>
	);
	const page = getByText(/Login Here/i);
	expect(page).toBeInTheDocument();
});
