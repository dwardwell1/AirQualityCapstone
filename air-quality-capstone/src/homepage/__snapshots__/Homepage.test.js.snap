// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Homepage pt-0 "
  >
    <div
      class="container text-center"
    >
      <h1
        class="mb-4 font-weight-bold"
      >
        Current Air Quality Search
      </h1>
      <p
        class="lead"
      >
        Check your local air quality.
      </p>
      <div
        class="LoginForm"
      >
        <div
          class="container col-md-6 offset-md-3 col-lg-4 offset-lg-4"
        >
          <h3
            class="mb-3"
          >
            AQI based on ZIP code
          </h3>
          <div
            class="card "
          >
            <div
              class="card-body"
            >
              <form>
                <div
                  class="form-group"
                >
                  <label>
                    ZIP code
                  </label>
                  <input
                    autocomplete="zipcode"
                    class="form-control"
                    name="zipcode"
                    required=""
                    value=""
                  />
                </div>
                <button
                  class="btn btn-primary float-right"
                >
                  Submit
                </button>
              </form>
            </div>
          </div>
        </div>
      </div>
      <h5
        class="mt-5"
      >
        View 
        <a
          href="/subs"
        >
          Subscribed Location
        </a>
         AQ 
      </h5>
    </div>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot when logged out 1`] = `
<DocumentFragment>
  <div
    class="Homepage pt-0 "
  >
    <div
      class="container text-center"
    >
      <h1
        class="mb-4 font-weight-bold"
      >
        Current Air Quality Search
      </h1>
      <p
        class="lead"
      >
        Check your local air quality.
      </p>
      <div
        class="LoginForm"
      >
        <div
          class="container col-md-6 offset-md-3 col-lg-4 offset-lg-4"
        >
          <h3
            class="mb-3"
          >
            AQI based on ZIP code
          </h3>
          <div
            class="card "
          >
            <div
              class="card-body"
            >
              <form>
                <div
                  class="form-group"
                >
                  <label>
                    ZIP code
                  </label>
                  <input
                    autocomplete="zipcode"
                    class="form-control"
                    name="zipcode"
                    required=""
                    value=""
                  />
                </div>
                <button
                  class="btn btn-primary float-right"
                >
                  Submit
                </button>
              </form>
            </div>
          </div>
        </div>
      </div>
      <p>
        <a
          class=" font-weight-bold"
          href="/signup"
        >
          Sign up
        </a>
         for Email alerts or
        <br />
        <a
          class=" font-weight-bold"
          href="/login"
        >
          Login here
        </a>
      </p>
    </div>
  </div>
</DocumentFragment>
`;
